tools/c7n_azure/c7n_azure/functionapp_utils.py

```diff
@@ -71,9 +91,110 @@ def deploy_dedicated_function_app(parameters):
         storage_account_id = sa_unit.provision_if_not_exists(parameters.storage_account).id
         con_string = FunctionAppUtilities.get_storage_account_connection_string(storage_account_id)
 
-        function_app_params.update({'location': app_service_plan.location,
-                                    'app_service_plan_id': app_service_plan.id,
-                                    'app_insights_key': app_insights.instrumentation_key,
-                                    'storage_account_connection_string': con_string})
+        function_app_params.update(
+            {'location': app_service_plan.location,
+             'app_service_plan_id': app_service_plan.id,
+             'app_insights_key': app_insights.instrumentation_key,
+             'is_consumption_plan': FunctionAppUtilities.is_consumption_plan(parameters),
+             'storage_account_connection_string': con_string})
 
         return function_app_unit.provision(function_app_params)
+
+    @classmethod
+    def publish_functions_package(cls, function_params, package):
+        session = local_session(Session)
+        web_client = session.client('azure.mgmt.web.WebSiteManagementClient')
+
+        cls.log.info('Publishing Function application')
+
+        # provision using Kudu Zip-Deploy
+        if not cls.is_consumption_plan(function_params):
+            publish_creds = web_client.web_apps.list_publishing_credentials(
+                function_params.function_app_resource_group_name,
+                function_params.function_app_name).result()
+
+            if package.wait_for_status(publish_creds):
+                package.publish(publish_creds)
+            else:
+                cls.log.error("Aborted deployment, ensure Application Service is healthy.")
+        # provision using WEBSITE_RUN_FROM_PACKAGE
+        else:
+            # fetch blob client
+            blob_client = StorageUtilities.get_blob_client_from_storage_account(
```

Comment:
Does that represent a new role they need on their SP or would it have already been there? Blob writer or whatever it is called...

Reply:
nm, I see its just a list keys call.

Reply:
nm I see its just a list keys call.